### PR TITLE
Gamemode dependent block outline rendering.

### LINKED ISSUE #137: Don't render targeted block outline when in Spectator mode
Targeted block outlines shouldn't be rendered while in spectator mode. The relevant code is in `WorldRenderer`'s `render` function.

### PR DESCRIPTION
**Pull request by gracien-app**
# Description

This PR fixes the case where blocks are given target outlines when player's current gamemode is Spectator mode.

Block targetability is determined in function returning targeted block position. Returning `nil` value in aforementioned function naturally prevents rendering of block outline in `worldRenderer` main `render` loop (no block position data available).

Initial approach was to use computed variable in `worldRender`, however, proposed solution seemed much more elegant (single thread-safe player data fetch).

Fixes #137 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation comments
- [x] My changes generate no new warnings

### CODE REVIEW

File: Sources/Core/Sources/Game.swift
Comment (stackotter) [inline Sources/Core/Sources/Game.swift:270]: I think the gamemode check should be in the renderer instead of the `targetedBlock` method because in spectator I would still expect a targeted block to be returned (for f3 screen purposes), it just wouldn't have an outline rendered

**Review by stackotter — CHANGES_REQUESTED** [review]
Seems to work well, just one change requested to make the API behave more how I would expect.

File: Sources/Core/Sources/Game.swift
Comment (gracien-app) [reply Sources/Core/Sources/Game.swift]: Fair enough. I haven't seen any usage of this function outside of `worldRenderer` which is why i went with this, a little bit destructive, approach. Corrected.

File: Sources/Core/Sources/Render/World/WorldRenderer.swift
Comment (stackotter) [inline Sources/Core/Sources/Render/World/WorldRenderer.swift]: I think this check is unnecessary, the direct comparison with `.spectator` should work for an optional too

**Review by stackotter — CHANGES_REQUESTED** [review]
Just a small nit-picky change requested this time, nothing major. That'll be the last change (I swear actually this time), I'll merge once that's fixed.

File: Sources/Core/Sources/Render/World/WorldRenderer.swift
Comment (gracien-app) [reply Sources/Core/Sources/Render/World/WorldRenderer.swift]: Do you mean... the entire computed variable, doing `client.game.currentGamemode()` in `render()` or just getting rid of guard?

**Review by stackotter — APPROVED** [review]
Looks good to me 👍

**Comment by stackotter** [comment]
I'll just wait for the build action to finish before merging just in case

**Pull request merged**
